api/src/main/scala/za/co/absa/loginsvc/rest/config/provider/ConfigProvider.scala

```diff
@@ -74,7 +74,10 @@ class ConfigProvider(@Value("${spring.config.location}") yamlPath: String)
         ldapConfig.throwErrors()
         ldapConfig
       }
-    else ActiveDirectoryLDAPConfig("", "", "", 0, None)
+    else{
+      val emptyServiceAccount = ServiceAccountConfig("", Option(IntegratedLdapUserConfig("","")), None)
+      ActiveDirectoryLDAPConfig("", "", "", 0, emptyServiceAccount, None)
+    }
```

Comment:
This whole method seems a bit weird and more even so with defining an `emptyServiceAccount` here.

https://github.com/AbsaOSS/login-service/pull/101/files#diff-c1c2d4980084933489c41a4356ad8db2c69e005ccddf0fddf877e1d6c586b2e7R63 Documents that a service account must be given, so we should fail if not present -- I see no reason to create an `emptyServiceAccount`.

Also, on a more general note, I wonder - if the whole
```scala
trait AuthConfigProvider {
  def getLdapConfig: ActiveDirectoryLDAPConfig // todo Option?
  def getUsersConfig: UsersConfig // todo Option?
}
```
should not rather yield Option for both configs -- they are, IIRC, optional in their nature.